lib/src/builder/motion_animated_builder.dart

```diff
@@ -135,9 +135,9 @@ class MotionBuilderState extends State<MotionBuilder>
       ..sort();
 
     final motionData = MotionData(
-      endOffset: _itemOffsetAt(itemIndex) ?? Offset.zero,
-      startOffset: _itemOffsetAt(itemIndex) ?? Offset.zero,
-    );
+        endOffset: Offset.zero,
+        startOffset: Offset.zero,
+        duration: insertDuration);
```

Comment:
Why do we need to add duration in `MotionData`?

Reply:
There is no any option to detect whether the  item is added or deleted from the list and we can only set it in the map while inserting or deleting the item and we can set duration of the controller according to `MotionData.duration`.